File: front-end/cultural-content-client/src/app/models/cultural-offer-response.ts
import { CulturalOfferLocation } from './culutral-offer-location';
import { Location } from './culutral-offer-location';
import { CulturalOfferSubType } from './culutral-offer-subType';

export interface CulturalOfferResponse {
  _id?: any;
  name: String;
  description: String;
  location: Location;
  subType: CulturalOfferSubType;
  images: String[];
}

File: back-end/culture-content-app/src/main/java/com/example/culturecontentapp/payload/response/CulturalOfferResponse.java
package com.example.culturecontentapp.payload.response;

import java.util.Set;

import com.fasterxml.jackson.annotation.JsonProperty;

public class CulturalOfferResponse {

  @JsonProperty("id")
  private Long id;

  @JsonProperty("name")
  private String name;

  @JsonProperty("description")
  private String description;

  @JsonProperty("location")
  private CulturalOfferLocationResponse location;

  @JsonProperty("images")
  private Set<String> images;

  @JsonProperty("subType")
  private SubTypeResponse subType;

  protected CulturalOfferResponse() {

  }

  public Long getId() {
    return this.id;
  }

  public String getName() {
    return this.name;
  }

  public String getDescription() {
    return this.description;
  }

  public CulturalOfferLocationResponse getLocation() {
    return this.location;
  }

  public Set<String> getImages() {
    return this.images;
  }

  public SubTypeResponse getSubType() {
    return this.subType;
  }
}
